pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,214 @@
+import warnings
+
+import numpy as np
+import pymc as pm
+import pytensor
+import pytensor.tensor as at
+from pymc.distributions.dist_math import check_parameters
+from pymc.distributions.distribution import (
+    Distribution,
+    SymbolicRandomVariable,
+    _moment,
+)
+from pymc.distributions.logprob import ignore_logprob, logp
+from pymc.distributions.shape_utils import (
+    _change_dist_size,
+    change_dist_size,
+    get_support_shape_1d,
+)
+from pymc.logprob.abstract import _logprob
+from pymc.util import check_dist_not_registered
+from pytensor.graph.basic import Node
+from pytensor.tensor import TensorVariable
+from pytensor.tensor.random.op import RandomVariable
+
+
+class DiscreteMarkovChainRV(SymbolicRandomVariable):
+    default_output = 1
+    _print_name = ("DiscreteMC", "\\operatorname{DiscreteMC}")
+
+    def update(self, node: Node):
+        return {node.inputs[-1]: node.outputs[0]}
+
+
+class DiscreteMarkovChain(Distribution):
+    r"""
+    A Discrete Markov Chain is a sequence of random variables
+    .. math::
+        \{x_t\}_{t=0}^T
+    Where transition probability P(x_t | x_{t-1}) depends only on the state of the system at x_{t-1}.
+
+    Parameters
+    ----------
+    P: tensor
+        Matrix of transition probabilities between states. Rows must sum to 1.
+        One of P or P_logits must be provided.
+    P_logit: tensor, optional
+        Matrix of tranisiton logits. Converted to probabilities via Softmax activation.
+        One of P or P_logits must be provided.
+    steps: tensor, optional
+        Length of the markov chain. Only needed if state is not provided.
+    init_dist : unnamed distribution, optional
+        Vector distribution for initial values. Unnamed refers to distributions
+        created with the ``.dist()`` API. Distribution should have shape n_states.
+        If not, it will be automatically resized. Defaults to pm.Categorical.dist(p=np.full(n_states, 1/n_states)).
+        .. warning:: init_dist will be cloned, rendering it independent of the one passed as input.
+    """
```

Comment:
I suggest you add an example code snippet

Reply:
And an entry to the docs

Reply:
Where would I update the docs? I don't see anything in pymc-experimental/docs

Reply:
Here I think?

https://github.com/pymc-devs/pymc-experimental/blob/main/docs/api_reference.rst

Reply:
CC @OriolAbril

Reply:
the current api docs look a bit strange (and limited?). The main question is "how should users import this distribution?", then I'll be able to indicate how to document it.

Reply:
Strange how? I tried to follow the style in the other timeseries.py distributions -- eager to fix it if it's off the mark.

Reply:
> Strange how?

There are 3 modules _documented_, and each has only one object inside of it. Yet there are 2 section headings seemingly indicating the existence of 2 modules only which are a superset of the documented modules.

My hunch is that what should be documented are two modules (trusting the headings instead of the `automodule` directives), `distributions` and `utils` with elements being imported from there, not from their specific file (e.g. `pmx.utils.prior_from_idata` instead of `pmx.utils.prior.prior_from_idata`). It also seems clear that there are several objects missing from the API reference.

Reply:
There's an open issue to fix it: https://github.com/pymc-devs/pymc-experimental/issues/101

Should we not even try to include your distribution in the docs in this PR?

Reply:
We can already add it in the distributions section as pymc_experimental.distributions.timeseries.DiscreteMarkovChain

Reply:
What do I need to do to resolve this?

Reply:
Just list it here? https://github.com/pymc-devs/pymc-experimental/blob/main/docs/api_reference.rst#distributions